packager/packager.cc

```diff
@@ -1482,10 +1482,12 @@ void validatePackagedFile(core::Context ctx, const ast::ExpressionPtr &tree) {
     auto &pkg = ctx.state.packageDB().getPackageForFile(ctx, ctx.file);
     if (!pkg.exists()) {
         // Don't transform, but raise an error on the first line.
-        if (auto e = ctx.beginError(core::LocOffsets{0, 0}, core::errors::Packager::UnpackagedFile)) {
-            e.setHeader("File `{}` does not belong to a package; add a `{}` file to one "
-                        "of its parent directories",
-                        ctx.file.data(ctx).path(), PACKAGE_FILE_NAME);
+        if (!pkg.ignoreForPackaging()) {
+            if (auto e = ctx.beginError(core::LocOffsets{0, 0}, core::errors::Packager::UnpackagedFile)) {
+                e.setHeader("File `{}` does not belong to a package; add a `{}` file to one "
+                            "of its parent directories",
+                            ctx.file.data(ctx).path(), PACKAGE_FILE_NAME);
```

Comment:
What do you think about an alternative implementation where we still require all files to be in a package, so you would have to have a `__package.rb` file somewhere in your `sorbet/rbi/` folder, but we make the option treat the files as `# packaged: false` if they're inside those directories? Or alternatively, we allow specifying this in the `__package.rb` file itself instead of command line options?

The refactors that I'm working on to move the package DB into the symbol table assume that every file has a package.